modules/core/include/opencv2/core/hal/hal.hpp

```diff
@@ -50,6 +50,15 @@
 #include "opencv2/core/hal/interface.h"
 
 //! @cond IGNORED
+#define CALL_HAL_RET(name, fun, retval, ...) \
```

Comment:
Since these macros are only used to make internal calls to HAL, I suggest to move them to `core/src/hal_replacement.hpp` and a copy to `imgproc/src/hal_replacement.hpp` (or some other private header).